### DISCUSSION TITLE
Exception Handling Takes too long

### DISCUSSION THREAD

**Discussion by raqibhayder** [fastapi/fastapi#3951] [Questions]
I have a FastAPI application running in production which runs fine for the most part, but there are bursts where the requests take too long to execute. 

Below is the Transaction Trace of the request. What I do not understand is why the exception handling takes 19 seconds to complete. Any pointers or direction would be much appreciated.

![image](https://user-images.githubusercontent.com/1992290/134939492-ecd701f6-e476-43dc-96fe-690c95ce8ef9.png)

---

I have cross posted this issue in Starlette discussion as well. [source](https://github.com/encode/starlette/discussions/1294)

**Comment by ghandic** [comment]
I suggest you add more detail into the starlette discussion post and follow it up on there as it is below the stack of FastAPI.

Why are you having 19,100 exceptions in the first place out of interest?

**Reply by raqibhayder** [reply]
Thank you for responding. 

> Why are you having 19,100 exceptions in the first place out of interest?

I am not having 19,100 exceptions. The time it took to execute that section of the code took 19.1 seconds. I could not figure out why that is happening.

**Reply by ghandic** [reply]
Have you tried running this in debug and stepping into the exception to see where it is getting to? It does seem odd, it looks like its waiting for something... a rollback or something? Do you get the same time lag when you remove postgres from the equation?

**Comment by ajay-jax** [comment]
whats the workaround or RCA for this issue??

**Reply by ajay-jax** [reply]
i am also facing same

**Comment by AlanCanalesM** [comment]
Is there any solution for this issue? im still facing the same

**Reply by YuriiMotov** [reply]
@AlanCanalesM, could you provide an [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example), so that we could reproduce this?
